pkg/types/restorer.go

```diff
@@ -61,7 +61,8 @@ type RestoreOptions struct {
 type RestorationConfig struct {
 	InitialCluster           string   `json:"initialCluster"`
 	InitialClusterToken      string   `json:"initialClusterToken,omitempty"`
-	RestoreDataDir           string   `json:"restoreDataDir,omitempty"`
+	DataDir                  string   `json:"dataDir,omitempty"`
+	TempDir                  string   `json:"tempDir,omitempty"`
```

Comment:
`TempDir` is a field within `RestorationConfig`, so it still implies that it's a temporary directory used during restoration. I don't feel we need to expand the variable name any more than that. WDYT?

Reply:
TempDir is too generic IMHO. It is possible to use it for anything. However you have a very specific purpose for it and that is for storing delta snapshots. Ideally it is not even required to create a command line flag for this as then it leaks implementation details to the consumer.

Reply:
When the user is setting the `snapstore-temp-directory` to set where to store temporary snapshots before pushing them to the bucket, why not allow them to set `restoration-temp-dir` as well? 

Also, only the user has full knowledge about the filesystem on which they're running etcdbrctl, and the code cannot make any assumptions about the directories that are present (or not present). This was one of the reasons we removed `/tmp` as the default temp dir for storing delta snapshots during restoration. When the user can define where to store these temp files, it gives them more determinism and control over what parts of the filesystem are being used for what, and makes it easier to debug potential disk I/O or bloating issues as well.

I feel `RestorationConfig.TempDir` is sufficient, since it denotes that it is a temporary directory used during restoration, and does not overstep the purpose of `RestorationConfig.DataDir` which is used for the actual etcd data being restored. If you still insist, we can probably call it `TempDeltaSnapshotsDir`. WYDT?

Reply:
>Also, only the user has full knowledge about the filesystem on which they're running etcdbrctl, and the code cannot make any assumptions about the directories that are present (or not present)

The base path is always standard path that will always be present in all linux distros. The subdirectories are always created. Currently we do not anyways support non-linux containers. 

If you wish to have these configurable then the purpose of a flag should be very clear. If they wish to you a mount for that then they have to appropriately size the mount. Temp directory does not tell me anything. So from command line argument perspective it should clearly indicate what it is used for. From exposed struct/variable/function/interface perspective it should be very clear on what it is meant for.

This is just my view. But you have the freedom to overwrite it and go with `TempDir` as well.

Reply:
We can change it to `TempDeltaSnapshotsDir` then. And the flag would be `restoration-temp-delta-snapshots-dir`. Is that ok with you?

Reply:
Yes that is fine. You could also make it shorter by just using `TempSnapshotsDir` in the `RestorationConfig` struct and similarly `restoration-snapshot-temp-dir` But only if you wish to make it smaller. No hard views on it.

Reply:
Sure, I'll make the change then